src/admin/admin.js

```diff
@@ -11,13 +11,13 @@ import { Redirect } from 'react-router-dom';
 export default function Admin(props) {
     const [existedHubs, setExistedHubs] = useState([]);
 
-    if (props.data.ifLoggedIn == false) {
+    if (Boolean(props.data.ifAdmin) === false) {
```

Comment:
if you put all authorizating data to session, why you need pass them through props? You can directly retrieve them from session.